src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+
+import java.io.*;
+
 public class WorkWithFile {
+    int buySum = 0;
+    int supplySum = 0;
     public void getStatistic(String fromFileName, String toFileName) {
+        readFromFile(fromFileName);
+        calculateReport();
+        writeToFile(toFileName);
+        cleanUp();
+    }
+
+    private void cleanUp() {
+        buySum = 0;
+        supplySum = 0;
+    }
+
+    public void readFromFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))){
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                String type = parts[0];
+                int amount = Integer.parseInt(parts[1]);
+
+                if ("supply".equals(type)) {
+                    supplySum += amount;
+                } else if ("buy".equals(type)) {
+                    buySum += amount;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't find file ", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file ", e);
+        }
```

Comment:
The method 'readFromFile' should return the calculated values instead of modifying the class fields directly. This will allow you to make 'buySum' and 'supplySum' local variables.